src/psyclone/psyir/nodes/literal.py

```diff
@@ -74,7 +74,7 @@ class Literal(DataNode):
     _children_valid_format = "<LeafNode>"
     _text_name = "Literal"
     _colour = "yellow"
-    _real_value = r'^[+-]?[0-9]+(\.[0-9]*)?(e[+-]?[0-9]+)?$'
+    _real_value = r'^[+-]?[0-9]+(\.[0-9]*)?([eE][+-]?[0-9]+)?$'
```

Comment:
Now that I notice it, I believe line 105 should also be a `re.match` instead of a `re.search`

Reply:
Also, can you double check if in C/C++ an upper case E is valid for scientific notation. If it does no problem, otherwise we have to account for this in the backend, or revert the change and do a self.name.lower() in the psyir_expression check.

Reply:
Yes, confirmed with gcc and icc, and https://docs.microsoft.com/en-us/cpp/c-language/c-floating-point-constants?view=msvc-160, which specifies lower and upper case for `e` 😄